djstripe/contrib/rest_framework/serializers.py

```diff
@@ -8,12 +8,197 @@
+    start = serializers.DateTimeField(required=False)
+    start_date = serializers.DateTimeField(required=False)
+    status = serializers.CharField(required=False)
+    tax_percent = serializers.DecimalField(
+        required=False, max_digits=5, decimal_places=2
+    )
+    trial_end = serializers.DateTimeField(required=False)
+    trial_start = serializers.DateTimeField(required=False)
+
+    def validate_status(self, value):
+        if value not in dir(SubscriptionStatus):
+            raise ValidationError(
+                {"detail": "Invalid SubscriptionStatus {}".format(value)}
+            )
+        return value
+
+    def create(self, validated_data):
+        raise MethodNotAllowed("POST")
+
+    def update(self, instance: Subscription, validated_data: dict):
+        # We use UPDATE instead of DELETE to cancel a subscription, since
+        # cancelling means change the internal state, but not remove it from the DB.
+
+        status = validated_data.get("status")
+
+        # It is a usual ambiguity of expressing an ACTION through REST APIs which
+        # are fundamentally based on manipulating resources.
+        if status == SubscriptionStatus.canceled != instance.status:
+            try:
+                instance.cancel(at_period_end=CANCELLATION_AT_PERIOD_END)
+            except Exception as e:
+                msg = "Something went wrong cancelling the subscription: " + str(e)
+                raise ValidationError(detail=msg)
+
+        # Applying the update of all other fields.
+        Subscription.objects.filter(pk=instance.pk).update(**validated_data)
+        instance.refresh_from_db()
+        return instance
+
+
+class CreateSubscriptionSerializer(SubscriptionSerializer):
+    """Extend the standard SubscriptionSerializer for the case of creation,
+    which must include a stripe_token field, although it doesn't belong
+    to the model itself."""
+
+    # Required on creation
+    plan = serializers.SlugField(required=True, source="plan.id")
+    stripe_token = serializers.CharField(max_length=200, required=True)
+    charge_immediately = serializers.NullBooleanField(required=False, default=True)
+
+    def validate(self, attrs):
+        # Since we use source=plan.id in field declaration, one must overcome its
+        # nested nature when validating the final data. This can't be done in a
+        # specialized validate_plan function (where the value will appear as provided
+        # in POST request, that is, the correct ID slug string.)
+        plan_id = attrs.pop("plan").get("id")
+
+        if Plan.objects.filter(id=plan_id).exists():
+            attrs.update(plan=Plan.objects.get(id=plan_id))
+        else:
+            msg = "Unknown / invalid Plan id: " + str(attrs.get("plan"))
+            raise ValidationError(detail=msg)
+
+        return attrs
+
+    def create(self, validated_data: dict):
+        stripe_token = validated_data.pop("stripe_token")
+        self.customer.add_card(stripe_token)
```

Comment:
as noted elsewhere, there can be multiple `customer` for a single user.
however the plan has a `djstripe_owner_account` , and "user + plan.djstripe_owner_account" can be used to get a unique customer.